src/runtime_src/core/tools/xbmgmt2/flash/flasher.h

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (C) 2018 Xilinx, Inc
+ * Author: Ryan Radjabi
+ *
+ * This is a wrapper class that does the prep work required to program a flash
+ * device. Flasher will create a specific flash object determined by the program
+ * mode read from FeatureROM. Common functions between XSPI_Flasher and BPI_Flasher
+ * are implemented here.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef FLASHER_H
+#define FLASHER_H
+
+#include "xspi.h"
+#include "xmc.h"
+#include "firmware_image.h"
+#include "xclfeatures.h"
+#include <sys/stat.h>
```

Comment:
How does this compile on windows?

Reply:
Not sure... but it does. Let me clean up the imports before you merge